### DISCUSSION TITLE
Configure to not update Oh my bash.

### DISCUSSION THREAD

**Discussion by josys36** [topgrade-rs/topgrade#752] [Q&A] [answered]
Every time I run topgrade it tries to update oh my bash. How do I configure so it skips it? I tried to exclude oh my bash but it complains that's not a valid value to exlude.

**Comment by SteveLauC** [comment, marked as answer]
OMB belongs to step `shell`, so try:

```sh
$ topgrade --disable shell
```


An unfortunate thing is, OMB is not the only tool that belongs to `shell`, so disabling `shell` would also skip other `shell` tools.

**Reply by josys36** [reply]
That took care of it. OMB is the only shell tool I have installed, or at least I've never seen Topgrade update anything else in that catagory, so I'm OK with it.

**Reply by ylluminate** [reply]
It would actually be very nice to have a way to be more fine-grained with such updates, eg `--disable shell=omb,somethingelse` or such. The current shotgun approach is a bit too categorical... Same is true for `containers` as it should be possible to isolate different containerization systems...

**Reply by SteveLauC** [reply]
> It would actually be very nice to have a way to be more fine-grained with such updates, eg --disable shell=omb,somethingelse or such. The current shotgun approach is a bit too categorical

I agree, see #515.

> Same is true for containers as it should be possible to isolate different containerization systems...

Would you mind elaborating on this a bit?

**Reply by ylluminate** [reply]
The current `--disable shell` option, while functional, ends up being too broad and, as you have agreed, in setups where multiple shell tools are in use—like Oh My Zsh, Prezto, or even just minimal Zsh customizations—it would be helpful to selectively disable updates on one (or more) of those tools without impacting the others. This way, users can avoid potential breakage or unnecessary updates in specific shell environments they don’t want to touch... This much is, of course obvious.

Similarly, in the containerization context, we might be using different container runtimes or services like Docker, Podman (which I think are the only two at present supported?), or containerd, and being able to specify which to exclude/update could help prevent some problems—or enhance control (which is what I'm thinking in my case). It’s especially useful in environments where isolation of services is critical.

Does that make sense?

**Reply by SteveLauC** [reply]
Topgrade 16 allows you to choose the container runtime to use, and only one runtime can be used, does this work in your case?

**Reply by ylluminate** [reply]
Hmm. That might be sufficient. I suppose we could have multiple containerizations going. Hmmm.

**Reply by SteveLauC** [reply]
> I suppose we could have multiple containerizations going. 

Technically, considering that Docker and Podman do not overlap, this is possible to implement this. But I don't see the reason why a person would bother with both. 🤔

**Reply by ylluminate** [reply]
In my case I hate Docker, but I have to use it. I've been thinking about moving some things to Podman, so I could see myself having a (an indefinite) period of overlap... Various permutations could exist in this area as things evolve.

**Reply by SteveLauC** [reply]
> In my case I hate Docker, but I have to use it.

Get it, perhaps Topgrade should implement this feature:

> I suppose we could have multiple containerizations going.
